temporalcli/strings.go

```diff
@@ -33,6 +34,32 @@ func (s *StringEnum) Set(p string) error {
 
 func (*StringEnum) Type() string { return "string" }
 
+type StringEnumArray struct {
+	Allowed []string
+	Values  []string
+}
+
+func NewStringEnumArray(allowed []string, values []string) StringEnumArray {
+	// store allowed values in lower case so we can do case-insensitive comparison
+	for i, str := range allowed {
+		allowed[i] = strings.ToLower(str)
+	}
+
+	return StringEnumArray{Allowed: allowed, Values: values}
+}
+
+func (s *StringEnumArray) String() string { return strings.Join(s.Values, ",") }
+
+func (s *StringEnumArray) Set(p string) error {
+	if !slices.Contains(s.Allowed, strings.ToLower(p)) {
+		return fmt.Errorf("invalid value: %s, allowed values are: %s", p, strings.Join(s.Allowed, ", "))
+	}
+	s.Values = append(s.Values, p)
```

Comment:
maybe this should canonicalize what the user typed to the "correct" case? easy way would be make Allowed into a map from lower to correct, and then check by looking up